ui/pages/confirmations/components/confirm/info/hooks/useIsUpgradeTransaction.ts

```diff
@@ -3,13 +3,18 @@ import { Hex } from '@metamask/utils';
 import { useConfirmContext } from '../../../../context/confirm';
 import { EIP_7702_REVOKE_ADDRESS } from '../../../../hooks/useEIP7702Account';
 
-export function useIsUpgradeTransaction(): boolean {
+export function useIsUpgradeTransaction() {
   const authorizationAddress = useTransactionAuthorizationAddress();
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+  const { nestedTransactions } = currentConfirmation ?? {};
 
-  return (
+  const isUpgrade =
     Boolean(authorizationAddress) &&
-    authorizationAddress !== EIP_7702_REVOKE_ADDRESS
-  );
+    authorizationAddress !== EIP_7702_REVOKE_ADDRESS;
+  return {
+    isUpgrade,
+    isUpgradeWithNoNestedTransactions: isUpgrade && !nestedTransactions?.length,
```

Comment:
Would this be more explicit as as `isUpgradeOnly`?

Plus we could theoretically do an upgrade on any normal transaction so ideally would we check that `data` is empty or `0x` as that would include the standard and batch case.

Reply:
I updated the PR to address this